http-server: add page (#7200)
tldr-pages/tldr: pages/common/http-server.md

# http-server

> Simple static HTTP server to serve static files.
> More information: <https://github.com/http-party/http-server>.

> - Start a HTTP server listening on the default port to serve the current directory:

Reviewer: ```suggestion
- Start an HTTP server listening on the default port to serve the current directory:
```
This should be applied everywhere

Reviewer: @navarroaxel please apply this everywhere

`http-server`

> - Start a HTTP server on a specific port to serve files from a specific directory:

Reviewer: It's basically the same, but you used this wording in the example above
```suggestion
- Start a HTTP server on a specific port to serve a specific directory:
```

`http-server {{path/to/directory}} --port {{port}}`

- Start a HTTP server using basic authentication:

`http-server --username {{username}} --password {{password}}`

- Start a HTTP server with directory listings disabled:

`http-server -d {{false}}`

> - Start a HTTPS server on the default port using the specified certificates:

Reviewer: It's just one certificate in the example
```suggestion
- Start a HTTPS server on the default port using the specified certificate:
```

> - Start a HTTP server on the default port with SSL using the specified certificates:

Reviewer 2: Wording is technically wrong here.
```suggestion
- Start a HTTPS server on the default port using the specified certificates:
```
I don't like the fact that it uses the `--ssl` argument (since SSL was replaced by TLS ages ago), but there's nothing we can do about that.

`http-server --ssl --cert {{path/to/cert.pem}} --key {{path/to/key.pem}}`

> - Start a HTTP server and include the client's IP in the output logging:

Reviewer 2: Minor clarity
```suggestion
- Start a HTTP server and include the client's IP address in the output logging:
```

`http-server --log-ip`

> - Start a HTTP server with CORS enabled via the `Access-Control-Allow-Origin: *` header:

Reviewer 2: Perhaps this will read better?
```suggestion
- Start a HTTP server with CORS enabled by including the `Access-Control-Allow-Origin: *` header in all responses:
```

Reviewer 2: Any reason why you didn't apply this suggestion @navarroaxel?

Author: I've tested with the header applies only to `OPTIONS` methods or `GET` too.

`http-server --cors`

> - Start a HTTP server with log messages disabled:

Reviewer 2: Perhaps this will read better?
```suggestion
- Start a HTTP server with logging:
```

Reviewer: It should be
```suggestion
- Start a HTTP server with logging disabled:
```
or
```suggestion
- Start a HTTP server without logging:
```
Even tough, I don't have a problem with the original either.
